packages/cli/test/unit/cmds/validator/keymanager/keystoreCache.test.ts

```diff
@@ -0,0 +1,89 @@
+import fs from "node:fs";
+import {randomBytes} from "node:crypto";
+import tmp from "tmp";
+import {expect} from "chai";
+import chainAsPromised from "chai-as-promised";
+import chai from "chai";
+import {Keystore} from "@chainsafe/bls-keystore";
+import {interopSecretKey} from "@lodestar/state-transition";
+import bls from "@chainsafe/bls";
+import {SignerLocal, SignerType} from "@lodestar/validator";
+import {loadKeystoreCache, writeKeystoreCache} from "../../../../../src/cmds/validator/keymanager/keystoreCache.js";
+import {LocalKeystoreDefinition} from "../../../../../src/cmds/validator/keymanager/interface.js";
+
+chai.use(chainAsPromised);
+
+const numberOfSigners = 10;
+
+describe("keystoreCache", () => {
+  let definitions: LocalKeystoreDefinition[];
+  let signers: SignerLocal[];
+  let secretKeys: Uint8Array[];
+  let passwords: string[];
+  let keystoreCacheFile: string;
+  // tmp.setGracefulCleanup();
+
+  beforeEach(async function setup() {
+    this.timeout(50000);
+    definitions = [];
+    signers = [];
+    secretKeys = [];
+    passwords = [];
+    keystoreCacheFile = tmp.tmpNameSync({postfix: ".cache"});
+
+    for (let i = 0; i < numberOfSigners; i++) {
+      const secretKey = bls.SecretKey.fromBytes(interopSecretKey(i).toBytes());
+      const keystorePath = tmp.tmpNameSync({postfix: ".json"});
+      const password = secretKey.toHex();
+      const keystore = await Keystore.create(
+        password,
+        secretKey.toBytes(),
+        secretKey.toPublicKey().toBytes(),
+        keystorePath,
+        "test-keystore",
+        // To make the test efficient we use a low iteration count
+        {
+          function: "pbkdf2",
+          params: {dklen: 32, c: 10, prf: "hmac-sha256", salt: randomBytes(32).toString("hex")},
+        }
+      );
+      fs.writeFileSync(keystorePath, keystore.stringify());
+
+      signers.push({type: SignerType.Local, secretKey});
+
+      // Use secretkey hex as password
+      definitions.push({password: secretKey.toHex(), keystorePath});
+      passwords.push(password);
+      secretKeys.push(secretKey.toBytes());
+    }
+  });
+
+  describe("writeKeystoreCache", () => {
+    it("should write a valid keystore cache file", async () => {
+      await expect(writeKeystoreCache(keystoreCacheFile, signers, passwords)).to.fulfilled;
+      expect(fs.existsSync(keystoreCacheFile)).to.be.true;
```

Comment:
is there an easy way to do some basic sanitiy checks on the written file? I guess the proper check happens in the others test were the file is wirrten and the loaded, so maybe should not bother too much here

Reply:
The authenticity of files could better be checked in e2e tests.